GitCommands/ArgumentBuilderExtensions.cs

```diff
@@ -178,6 +178,31 @@ string GetArgument()
             }
         }
 
+        /// <summary>
+        /// Adds the git argument syntax for members of the <see cref="CleanMode"/> enum.
+        /// </summary>
+        /// <param name="builder">The <see cref="ArgumentBuilder"/> to add arguments to.</param>
+        /// <param name="mode">The enum member to add to the builder.</param>
+        public static void Add(this ArgumentBuilder builder, CleanMode mode)
+        {
+            builder.Add(GetArgument());
+
+            string GetArgument()
+            {
+                switch (mode)
+                {
+                    case CleanMode.OnlyNonIgnored:
+                        return "";
+                    case CleanMode.OnlyIgnored:
+                        return "-X";
+                    case CleanMode.All:
+                        return "-x";
+                    default:
+                        throw new InvalidEnumArgumentException(nameof(mode), (int)mode, typeof(CleanMode));
+                }
+            }
+        }
```

Comment:
Please a test for this method.

Reply:
I cannot see any reasonable test here, the logic is very simple. 
A test would require setting up FormCleanupRepository, we do not have any such tests right now.
It is possible to setup a provider and test that will just add another layer.

Reply:
Kind of why I wanted you to switch the enum to start with 0;

CleanMode mode;//defaults to 0 = sane default

Reply:
This is about testing GetCleanMode() not about testing CleanMode (that is done, adds no value in my opinion but just duplicate implementation so I comply). CleanMode has no default in the API, must be provided, similar to the other ArgumentBuilder extensions. (The bools here previously did not bother me but fair to change the implementation, I had done that for ResetMode already even for other reason: Remove code.) 
What to test here?

Reply:
Handled in #6002 
No test for GetCleanMode() 
Added test for all ArgumentBuilderExtensions enum